scripts/gas-refund-program/computeGasRefund.ts

```diff
@@ -55,6 +59,13 @@ async function startComputingGasRefundAllChains() {
           epoch <= epochInfo.getCurrentEpoch();
           epoch++
         ) {
+          if (GRPSystemGuardian.isMaxPSPGlobalBudgetSpent()) {
```

Comment:
Can there be a race conditions between different `isMax*` checks inside executions for different chains? Like while for one chain the max has been reached but execution on the other chain just passed one of the checks 1 sec before, so it continues without knowing the max has been reached.
Though you added these checks in multiple places, so maybe it won't happen.

Reply:
So the answer is yes to all you said. But in next iteration we'll have a bit more async steps this should be checked carefully.